PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -287,6 +289,15 @@ struct HfTrackIndexSkimCreatorTagSelTracks {
   Configurable<LabeledArray<double>> cutsTrackBach{"cutsTrackBach", {hf_cuts_single_track::cutsTrack[0], nBinsPtTrack, nCutVarsTrack, labelsPtTrack, labelsCutVarTrack}, "Single-track selections per pT bin for the bachelor of V0-bachelor candidates"};
   Configurable<double> etaMinTrackBach{"etaMinTrackBach", -99999., "min. pseudorapidity for bachelor in cascade candidate"};
   Configurable<double> etaMaxTrackBach{"etaMaxTrackBach", 0.8, "max. pseudorapidity for bachelor in cascade candidate"};
+  // soft pion cuts for D*
+  Configurable<double> ptMinSoftPionForDstar{"ptMinSoftPionForDstar", 0.05, "min. track pT for soft pion in D* candidate"};
+  Configurable<double> ptMaxSoftPionForDstar{"ptMaxSoftPionForDstar", 2., "max. track pT for soft pion in D* candidate"};
+  Configurable<double> etaMinSoftPionForDstar{"etaMinSoftPionForDstar", -99999., "min. pseudorapidity for soft pion in D* candidate"};
+  Configurable<double> etaMaxSoftPionForDstar{"etaMaxSoftPionForDstar", 0.8, "max. pseudorapidity for soft pion in D* candidate"};
+  Configurable<LabeledArray<double>> cutsTrackDstar{"cutsTrackDstar", {hf_cuts_single_track::cutsTrackPrimary[0], nBinsPtTrack, nCutVarsTrack, labelsPtTrack, labelsCutVarTrack}, "Single-track selections per pT bin for the soft pion of D* candidates"};
+  Configurable<bool> useIsGlobalTrackForSoftPion{"useIsGlobalTrackForSoftPion", false, "check isGlobalTrack status for soft pion tracks"};
+  Configurable<bool> useIsGlobalTrackWoDCAForSoftPion{"useIsGlobalTrackWoDCAForSoftPion", false, "check isGlobalTrackWoDCA status for soft pion tracks"};
+  Configurable<bool> useIsQualityTrackITSForSoftPion{"useIsQualityTrackITSForSoftPion", true, "check qualityTracksITS status for soft pion tracks"};
```

Comment:
I would prefer to keep them separated to have them more explicit (also keep in mind that we cannot filter the track table as in the example you linked, because we loop over track indices from the `trackToCollisionAssociator` task and not over tracks)

Reply:
The point was not to use filters, but just to have a single configurable with `0,1,2,...` options, which looks cleaner and less error prone than having multiple configurables to control the same thing (at the moment you have no checks on the configurations and you get undesired behavior if more than one of them is true). The help text is visible in the Hyperloop configuration, so it easy to understand what's happening if set properly

Reply:
In my opinion using arbitrary numbers is more error prone. At the moment there is no checks if more than one is enabled, but they are applied in a hierarchical order (i.e. if you have more than one enabled, always the tightest is applied, so there should be no undesired behaviour). Let's see what is the opinion of the others, but if we want to change, we have to change the convention also for the other tracks for consistency.

Reply:
So you say that `useIsGlobalTrack` is tighter than `isGlobalTrackWoDCA` which is tighter than `isQualityTrackITS`?
Also `itsClusterMap` is now used as a default selection if none of the configurables for the previous selections is set `true`. But if it can be only used for Run 2 converted data, it should not be the default, it should be separated from the others and checked against the `isRun2` value.

Reply:
> So you say that `useIsGlobalTrack` is tighter than `isGlobalTrackWoDCA` which is tighter than `isQualityTrackITS`?
Exactly. 

> Also `itsClusterMap` is now used as a default selection if none of the configurables for the previous selections is set `true`. But if it can be only used for Run 2 converted data, it should not be the default, it should be separated from the others and checked against the `isRun2` value.
I agree. 

If we want to modify this though, as mentioned in the reply to Fabio it should be modified also for the other tracks (not only the soft pion) for consistency. Personally I would completely remove the `doCutQuality` configurable (which is useless) and replace the "default" with something like `useQualityCutsRun2`. Regarding the single `int` configurable I have concerns about the fact that it's less error prone (for instance the description cannot be seen from the `json` itself, it appears only on hyperloop), but in any case if it is decided to do it, it has to be applied to all the types of tracks.

Reply:
I don't have a strong preference about 1 vs 3 configurables. Both options have their pros and cons and it does not prevent merging this PR, imho. If we change the selections, I would anyway prefer to do it in a separate PR.